tests/unit/src/test/scala/tests/DebugDiscoverySuite.scala

```diff
@@ -577,4 +593,756 @@ class DebugDiscoverySuite
       }
     }
   }
+
+  test("run-closest-test") {
+    val fileContent =
+      """|package a
+         |
+         |class Foo extends org.scalatest.funsuite.AnyFunSuite {
+         |  test("first") {}
+         |  test("second") {@@}
+         |}
+      """.stripMargin
+    val filePath = fooPath
+    val fileName = filePath.split("/").last
+    for {
+      _ <- initialize(
+        s"""/metals.json
+           |{
+           |  "a": {
+           |    "libraryDependencies":["org.scalatest::scalatest:3.2.16"]
+           |  }
+           |}
+           |/${filePath}
+           |${fileContent}
+           |""".stripMargin
+      )
+      _ <- server.didOpen(filePath)
+      (text, params) <- server.offsetParams(
+        fileName,
+        fileContent,
+        workspace,
+      )
+      _ <- server.didChange(filePath, text)
+      _ <- server.didSave(filePath)
+      _ <- server.waitFor(TimeUnit.SECONDS.toMillis(10))
+      debugger <- server.startDebuggingUnresolved(
+        new DebugDiscoveryParams(
+          server.toPath(filePath).toURI.toString,
+          "runClosest",
+          position = params.getPosition,
+        ).toJson
+      )
+      _ <- debugger.initialize
+      _ <- debugger.launch
+      _ <- debugger.configurationDone
+      _ <- debugger.shutdown
+      output <- debugger.allOutput
+    } yield assertNoDiff(
+      normalizeOutput(output),
+      """|Foo:
+         |- second
+         |1 tests, 1 passed
+         |All tests in a.Foo passed
+         |================================================================================
+         |All 1 test suites passed.
+         |================================================================================
+        """.stripMargin,
+    )
+  }
+
+  test("run-closest-suite") {
+    val fileContent =
+      """|package a
+         |
+         |class Foo extends org.scalatest.funsuite.@@AnyFunSuite {
+         |  test("first") {}
+         |  test("second") {}
+         |}
+         |""".stripMargin
+    val filePath = fooPath
+    val fileName = filePath.split("/").last
+    for {
+      _ <- initialize(
+        s"""/metals.json
+           |{
+           |  "a": {
+           |    "libraryDependencies":["org.scalatest::scalatest:3.2.16"]
+           |  }
+           |}
+           |/${filePath}
+           |${fileContent}
+           |""".stripMargin
+      )
+      _ <- server.didOpen(filePath)
```

Comment:
If a lot of the test cases are the same, I would extract this to check method with file structure and expected output as parameters